settings.gradle

```diff
@@ -92,7 +86,7 @@ include "spock-guice"
 include "spock-junit4"
 include "spock-testkit"
 
-def variant = System.getProperty("variant") as BigDecimal ?: 2.5
+def variant = providers.systemProperty("variant").getOrElse("2.5").toBigDecimal()
```

Comment:
The gradleProperty method was introduced in Gradle 6.2 and improves compatibility with Gradle's configuration cache.

Reply:
Not anymore.
It was necessary in the beginning to use those explicitly so that the configuration cache recognizes them as inputs.
But by now there was code added in Groovy that instruments the whole build script classpaths using bytecode manipulation to also capture all libraries and old plugins and whatever, that reads system properties, environment variables, files, calls external processes and so on.
So for the configuration cache to recognize them, those providers are also not necessary anymore.
Their main usecase now really is, if you need them as `Provider` to feed a `Property`.

Reply:
Revert it for now in 6b94ee1994050af2b0acc7667fa04d04ca2b0611.